staging/vhost-device-gpu/src/virtio_gpu.rs

```diff
@@ -610,6 +608,20 @@ impl VirtioGpu for RutabagaVirtioGpu {
             .get(&resource_id)
             .ok_or(ErrInvalidResourceId)?;
 
+        let rect = if self.renderer == GpuMode::ModeGfxstream {
+            // Gfxstream doesn't support transfer_read for portion of the resource. The width/height
+            // specified in stream_renderer_transfer_read_iov are ignored and the whole resource is
+            // copied.
+            Rectangle {
+                x: 0,
+                y: 0,
+                width: resource.width,
+                height: resource.height,
+            }
```

Comment:
I had a patch like this too in my local branch, I think this fix makesense.  same way it was handled in virtio-gpu-rutabaga. You can add a comment pointing to the issue where gfxstream stream_renderer_transfer_read_iov expects a certain width/height.